Grammar Updates to the General Folder (#1026)
AdguardTeam/AdguardKnowledgeBase: 01.general/15.dns-filtering/docs.en.md

<a id="dns-servers"></a>
#### DNS servers

> There are thousands of DNS servers to choose from, and they are all unique in their properties and purposes. Most simply return the IP address of the requested domain, but some have additional functions: they block ads, trackers, adult domains, and so on. Nowadays, all major DNS servers employ one or more reliable encryption protocols: DNS-over-HTTPS, DNS-over-TLS. AdGuard also provides a [DNS service](https://adguard-dns.com/en/welcome.html), and it was the world's first to offer the very new and promising [DNS-over-QUIC](https://adguard.com/en/blog/dns-over-quic.html) encryption protocol. AdGuard has different servers for different goals. This diagram illustrates how AdGuard blocking servers work:

Author: ```suggestion
There are thousands of DNS servers to choose from, and they are all unique in their properties and purposes. Most of them simply return the IP address of the requested domain, but some have additional features such as blocking ads, trackers, adult domains, and so on. Nowadays, all major DNS servers support such reliable encryption protocols as DNS-over-HTTPS or DNS-over-TLS. AdGuard has its own [DNS service](https://adguard-dns.com/en/welcome.html) and was the first provider in the world to offer a very new and promising [DNS-over-QUIC](https://adguard.com/en/blog/dns-over-quic.html) encryption protocol. AdGuard has different servers for different goals. This diagram illustrates how AdGuard's blocking servers work:
```

Network filtering is what we call the 'regular' way AdGuard standalone apps process network traffic, hence the name. Feel free to brush up on it by reading [this article](https://kb.adguard.com/en/general/how-ad-blocking-works).

First of all, we have to mention that with AdGuard you don't have to choose. You can always use both regular network filtering and DNS filtering at the same time. However, it's important to understand key differences between the two. DNS filtering has both unique advantages and drawbacks:

**Pros of DNS filtering:**

1. On some platforms, this is the only way to achieve system-wide filtering. For example, on iOS, only the Safari browser supports content blocking in the familiar sense, for everything else, there's only DNS filtering.
2. Some forms of tracking (like [CNAME-cloaked tracking](https://adguard.com/en/blog/cname-tracking.html)) can only be dealt with by DNS filtering.
3. The stage of processing a DNS request is the earliest you could possibly deal with an ad or a tracker. This helps save a little bit of battery life and traffic.

**Cons of DNS filtering:**

> 1. DNS filtering is "rough", meaning that it won't remove whitespaces that are left behind a blocked ad or apply any sort of cosmetic filtering. Many of the more complicated ads can't be blocked at the DNS level (or rather, they can, but only by blocking the entire domains that are being used for other purposes). 

Author: ```suggestion
1. DNS filtering is "rough": it doesn't remove whitespaces left behind by blocked ads or apply any cosmetic filtering. It is often impossible to block many of the more complex ads at the DNS level unless you block the entire domain that may be used for other purposes.
```
